VCServices/VCServices/VerifiableCredentialSDK.swift

```diff
@@ -5,26 +5,53 @@
 
 import VCEntities
 
+public enum VCSDKInitStatus
+{
+    case newMasterIdentifierCreated
+    case success
+}
+
 public class VerifiableCredentialSDK {
     
+    static let identifierService = IdentifierService()
+    
     /// Initialized the SDK.
     /// Returns: TRUE, if needed to create Master Identifier
     ///          FALSE, if Master Identifier is able to be fetched (included private keys from KeyStore)
     public static func initialize(logConsumer: VCLogConsumer = DefaultVCLogConsumer(),
-                                  userAgentInfo: String = "") throws -> Bool {
+                                  accessGroupIdentifier: String? = nil) -> Result<VCSDKInitStatus, Error> {
 
         VCSDKLog.sharedInstance.add(consumer: logConsumer)
-        VCSDKConfiguration.sharedInstance.setUserAgentInfo(with: userAgentInfo)
         
-        let identifierService = IdentifierService()
+        /// Get access group identifier for app.
+        if let accessGroupIdentifier = accessGroupIdentifier {
+            VCSDKConfiguration.sharedInstance.setAccessGroupIdentifier(with: accessGroupIdentifier)
+        }
         
+        /// Try to fetch master identifier from storage.
         do {
+            
             _ = try identifierService.fetchMasterIdentifier()
-            return false
+            
         } catch {
+            
+            /// If unable to fetch master identifier, create a new one.
             VCSDKLog.sharedInstance.logWarning(message: "Failed to fetch master identifier with: \(String(describing: error))")
-            _ = try identifierService.createAndSaveIdentifier(forId: VCEntitiesConstants.MASTER_ID, andRelyingParty: VCEntitiesConstants.MASTER_ID)
-            return true
+            return createNewIdentifier()
+            
+        }
+        
```

Comment:
Shouldn't we check here to see if the key can be read and fail if it's not? Then for the caller it's cleaner if the init succeed you know you're in a good state and if not then you can handle it differently based on the error returned

Reply:
oh yes thats what we wanted, good catch